### ISSUE TITLE
HPO search starts off with correct randomized settings then overwrites them when running

### ISSUE THREAD

**Issue by sirus20x6** [clearml/clearml#1464]
## Describe the bug
I create a base task, then try to run hpo. the task starts off with
optimizer_lr
optimizer_weight_decay
randomized like I asked, then as soon as it start training it replaces it with the base task lr and weight decay

## To reproduce
I use the gradio to select settings for my base run, I then hit the button to create an initial baseline, then I hit the buttons to run the hpo. I see the baseline register, the HPO job, then after a bit a new run starts with randomized parameters from the HPO, then they get overridden when training starts.

```
CONFIGURATION OBJECT
parameter_optimization_space = [{
    type = "LogUniformParameterRange"
    name = "General/optimizer_lr"
    min_value = -4.30102999566398
    max_value = -2.9999999999999996
    step_size = null
    include_max = true
    base = 10.0
  },
  {
    type = "LogUniformParameterRange"
    name = "General/optimizer_weight_decay"
    min_value = -1.9999999999999996
    max_value = -0.5228787452803376
    step_size = null
    include_max = true
    base = 10.0
  }]
```

## Expected behaviour
What is the expected behaviour? What should've happened but didn't?

## Environment
* Server type: self hosted docker. agent running locally
* ClearML SDK Version
* ClearML Server Version : WebApp: 2.2.0-690 • Server: 2.2.0-690 • API: 2.33
* Python Version 3.13
* OS arch linux

[gradio_ui.py](https://github.com/user-attachments/files/22180792/gradio_ui.py)

[clearml_utils.py](https://github.com/user-attachments/files/22180793/clearml_utils.py)

[hpo.py](https://github.com/user-attachments/files/22180824/hpo.py)

[train.py](https://github.com/user-attachments/files/22180835/train.py)

forgive my horrible code. I can't stand python and I'm not very good at it

**Comment by sirus20x6** [comment]
I'm not sure how, but I fixed it
